Docker updates

scytacki opened the pull request:
This includes a fix to prevent database corruption when running tests.
It also updates the docker documentation. The Readme has basic getting started instructions and there is a separate file for the details:
https://github.com/concord-consortium/rigse/blob/docker-updates2/doc/docker.md

pjanik commented on doc/docker.md:
```diff
@@ -0,0 +1,189 @@
+`docker-compose restart` which will fix things.  
+
+Also, when `config/database.yml` is not present yet, `docker-compose up` will copy it
+and run `rake db:setup`. `rake db:setup` erases any data in your database, so be careful.
```
It will be DB created by docker-compose, not the local one, so maybe we could mention that.

Also, all these things with files copying are confusing to me in the long term (even though it was me who I implemented that AFAIR). I never know what is done in my local filesystem and what is done in the container, and I had some issues with that. I guess that if you start from clean repo, it'll work as expected - the config files will be copied. But I'm not sure that if you remove those config files locally and run docker-compose again, it'll work as expected. I think the files will be there already.

scytacki replied:
Yeah this issue with copying is complex. Whether the files are there after you delete local copies will depend on several factors. I did make a change a little bit ago to make it a bit more deterministic.  Previously the files were added to the docker image built by Docker-dev.  I took that step out.  I'll test these instructions, and make sure I include all of the command needed to remove the files, so `docker-compose up`, does the right setup.

Alternatively we could try to change where Rails looks for these files, then we wouldn't need to copy them. However I think that might be confusing in a different way. 

Ideally we can remove the need for most of these files. Removing app_environment_variables would be easy. And it wouldn't take too much work to remove settings.yml.

scytacki commented:
I added the note about the database. I also added notes about reseting things if you are using unison. And finally I added a named mysql volume.
You will need to rm your mysql container before it starts using the new volume `docker-compose rm mysql`

scytacki commented:
@pjanik this latest commit should give you a docker-compose file that you can use to test production images.  That might not be exactly what you were trying to do before, but it seems like a good thing in any case.

If you want to mount and modify the files in the container then we need a slightly different compose file.  I think in that case the best approach is for the local (host) directory to start out empty and have docker (or unison) copy the files out of image into the directory. This should be easy to setup with either vanilla docker or unison.
